src/frequenz/channels/timer.py

```diff
@@ -604,8 +613,16 @@ def reset(self, *, start_delay: timedelta = timedelta(0)) -> None:
 
         if start_delay_ms < 0:
             raise ValueError(f"`start_delay` can't be negative, got {start_delay}")
-        self._stopped = False
+
+        if interval is not None:
+            self._interval = _to_microseconds(interval)
+
         self._next_tick_time = self._now() + start_delay_ms + self._interval
+
+        if self.is_running:
+            self._reset_event.set()
```

Comment:
Just out of curiosity, was the `if` necessary or is it just an *optimization*? Going through it quickly it looks like it should work without it too (if the timer is not running, it will just break the wait loop but it should resume normally, afterwards, right?), but maybe I'm missing some edge case.

Reply:
We would probably be fine if we just set it every time, yes.